docs/DeveloperGuide.md

```diff
@@ -302,28 +302,139 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+  
+        The use case ends.
+  
+  
+* 1d. The system detects an email input error (invalid format):
+    * 1d1. The system displays an error message stating the correct format
+  
+        The use case ends.
+  
+  
+* 1e. The system detects a duplicate phone number error:
+    * 1e1. The system displays an error message mentioning the existence of a duplicate phone number
+    
+        The use case ends.
+        
+
+* 1f. The system detects an invalid tag input:
+    * 1f1. The system displays an error message stating the tag is invalid
+    
+        The use case ends.
+
+
+
+**Use case: UC02 - Add Phone Number to Contact**
+
+**Guarantees:**
+* No duplicate phone numbers will exist across different contacts.
 
-  Use case ends.
 
-* 3a. The given index is invalid.
+**MSS**
+
+1.  User issues the add-phone command with a contact index and a phone number. 
+2.  The system adds the phone number to the contact and displays a success message
+3.  The system displays the updated contact information in the address book.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The system detects an invalid contact index:
+    * 1a1. The system displays an error message stating the contact index is invalid.
+
+        The use case ends.
+  
+  
+* 1b. The system detects a phone number input error (invalid format):
+    * 1b1. The system displays an error message stating the correct format
+    
+        The use case ends.
+  
+
+* 1c. The system detects a duplicate phone number error:
+    * 1c1. The system displays an error message mentioning the existence of a duplicate phone number
+    
+        The use case ends
+
+
+
+**Use case: UC03 - Add Address to Contact**
+
+**MSS**
+
+1.  User issues the add-address command with a contact index and an address. 
+2.  The system adds the address to the contact and displays a success message
+3.  The system displays the updated contact information in the address book.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The system detects an invalid contact index:
+    * 1a1. The system displays an error message stating the contact index is invalid.
+
+        The use case ends.
+  
+
+* 1b. The system detects an address input error (too long):
+    * 1b1. The system displays an error message stating the maximum length
+      
+        The use case ends.
+
+
+**Use case: UC04 - Add Email to Contact**
+
+**MSS**
+
+1.  User issues the add-email command with a contact index and an email address. 
+2.  The system adds the email address to the contact and displays a success message
+3.  The system displays the updated contact information in the address book.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The system detects an invalid contact index:
+    * 1a1. The system displays an error message stating the contact index is invalid.
+
+        The use case ends.
+
+
+* 1b. The system detects an email input error (invalid format):
+    * 1b1. The system displays an error message stating the correct format.
```

Comment:
I see, but in our case since we are referencing UC01 List all contacts use case,

> **Use case: UC01 -  List all contacts**
> 
> **MSS**
> 
> 1.  User issues the list command.
> 2.  The system retrieves and displays the list of all contacts to the user.
> 
>     Use case ends.

I think there should not be a repeat of step 2 saying the system displays a list of all contacts (Like in the example), in other use cases.

> **Use case: UC05 - Add Email to Contact**
> 
> **MSS**
> 
> 1. User <ins>lists all contacts (UC01)</ins>.
> 2. User issues the add-email command with the corresponding details. 
> 3. The system adds the email address to the contact and displays a success message
> 4. The system displays the updated contact information in the address book.

as such, I think in the extension it will look like this. I will clarify if this is correct, but for now this will be the changes before any further clarification is made. Thanks for pointing that out

> **Extensions**
> 
> * 1a. The list is empty.
> 
>     The use case ends.
> 
> 
> * 2a. The system detects an invalid contact index:
>     * 2a1. The system displays an error message stating the contact index is invalid.
> 
>         The use case resumes at step 2.
> 
> 
> * 2b. The system detects an email input error (invalid format):
>     * 2b1. The system displays an error message stating the correct format.
>   
>         The use case resumes at step 2.

Reply:
Yes, I agree, if we have that as UC01 then can definitely use that